src/pages/product-list-page/product-list-filter/product-list-filter.js

```diff
@@ -0,0 +1,328 @@
+import React, { useEffect, useState } from 'react';
+import FormControl from '@material-ui/core/FormControl';
+import FormGroup from '@material-ui/core/FormGroup';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import FormHelperText from '@material-ui/core/FormHelperText';
+import Checkbox from '@material-ui/core/Checkbox';
+import Typography from '@material-ui/core/Typography';
+import Slider from '@material-ui/core/Slider';
+import Paper from '@material-ui/core/Paper';
+import Button from '@material-ui/core/Button';
+import TextField from '@material-ui/core/TextField';
+
+import { useDispatch, useSelector } from 'react-redux';
+import useStyles from './product-list-filter.styles';
+import {
+  getFiltredProducts,
+  setColorsFilter,
+  setPatternsFilter,
+  setCategoryFilter,
+  setPriceFilter,
+  setSearchFilter
+} from '../../../redux/filter/filter.actions';
+
+export default function CheckboxesGroup({ selectedCategory }) {
+  const dispatch = useDispatch();
+
+  const styles = useStyles();
+
+  const {
+    products,
+    currentPage,
+    productsPerPage,
+    sortByPrice,
+    sortByRate,
+    sortByPopularity,
+    language
+  } = useSelector(
+    ({
+      Products: { products },
+      Filter: {
+        currentPage,
+        productsPerPage,
+        sortByPrice,
+        sortByRate,
+        sortByPopularity
+      },
+      Language: { language }
+    }) => ({
+      products,
+      currentPage,
+      productsPerPage,
+      sortByPrice,
+      sortByRate,
+      sortByPopularity,
+      language
+    })
+  );
+
+  useEffect(() => {
+    setPrice([
+      Math.min(...products.map((product) => product.basePrice)),
+      Math.max(...products.map((product) => product.basePrice))
+    ]);
+    setCategoryCheck({ [selectedCategory]: true });
+  }, [products, selectedCategory]);
+
+  const colors = [
+    ...new Set(
+      products.map((product) => JSON.stringify(product.colors[0].simpleName))
+    )
+  ].map(JSON.parse);
+  const categories = [
+    ...new Set(products.map((product) => JSON.stringify(product.category)))
+  ].map(JSON.parse);
+
+  const patterns = [
+    ...new Set(
+      products.map((product) =>
+        product.pattern.length
+          ? JSON.stringify(product.pattern)
+          : JSON.stringify([
+            {
```

Comment:
Please, move it to `translations` folder, and use here only imported variable of translation

Reply:
fixed